File: leetcode/2481-strictly-palindromic-number/solution.cpp
class Solution {
public:
    bool isStrictlyPalindromic(int n) {
        for (int i=2;i<n-1;i++){
            int num=n;
            vector<int> base;
            while(num!=0){
                base.push_back(num%i);
                num=num/i;
            }
            vector<int> revbase=base;
            reverse(base.begin(),base.end());
            for (int j=0;j<base.size();j++){
                if (base[j]!=revbase[j]){
                    return false;
                }
            }
        }
        return true;
    }
};

File: leetcode/2481-strictly-palindromic-number/solution.py
class Solution(object):
    def isStrictlyPalindromic(self, n):
        for i in range(2,n-1):
            base=[]
            num=n
            while num!=0:
                base.append(str(num%i))
                num//=i
            if base!=reversed(base):
                return False
        return True

        """
        :type n: int
        :rtype: bool
        """
        
